squidpy/pl/_spatial_utils.py

```diff
@@ -0,0 +1,897 @@
+    bg_size = (np.sqrt(gap_size) + (point * bg_width) * 2) ** 2
+    # the default black and white colors can be changes using the contour_config parameter
+    bg_color, gap_color = outline_color
+
+    if outline:
+        kwargs.pop("edgecolor", None)  # remove edge from kwargs if present
+        kwargs.pop("alpha", None)  # remove alpha from kwargs if present
+
+    return OutlineParams(outline, gap_size, gap_color, bg_size, bg_color), kwargs
+
+
+def _plot_scatter(
+    coords: NDArrayA,
+    ax: Axes,
+    outline_params: OutlineParams,
+    cmap_params: CmapParams,
+    color_params: Colormap,
+    size: float,
+    color_vector: NDArrayA,
+    **kwargs: Any,
+) -> Tuple[Axes, Collection | PatchCollection]:
+
+    if color_params.shape is not None:
+        scatter = partial(_shaped_scatter, shape=color_params.shape, alpha=color_params.alpha)
+    else:
+        scatter = partial(ax.scatter, marker=".", alpha=color_params.alpha, plotnonfinite=True)
+
+    if outline_params.outline:
+        _cax = scatter(
+            coords[:, 0],
+            coords[:, 1],
+            s=outline_params.bg_size,
+            c=outline_params.bg_color,
+            rasterized=sc_settings._vector_friendly,
+            cmap=cmap_params.cmap,
+            norm=cmap_params.norm,
+            **kwargs,
+        )
+        ax.add_collection(_cax)
+        _cax = scatter(
+            coords[:, 0],
+            coords[:, 1],
+            s=outline_params.gap_size,
+            c=outline_params.gap_color,
+            rasterized=sc_settings._vector_friendly,
+            cmap=cmap_params.cmap,
+            norm=cmap_params.norm,
+            **kwargs,
+        )
+        ax.add_collection(_cax)
+
+    _cax = scatter(
+        coords[:, 0],
+        coords[:, 1],
+        c=color_vector,
+        s=size,
+        rasterized=sc_settings._vector_friendly,
+        cmap=cmap_params.cmap,
+        norm=cmap_params.norm,
+        **kwargs,
+    )
+    cax = ax.add_collection(_cax)
+
+    return ax, cax
+
+
+def _plot_segment(
+    seg: NDArrayA,
+    cell_id: NDArrayA,
+    color_vector: NDArrayA | pd.Series[CategoricalDtype],
+    color_source_vector: pd.Series[CategoricalDtype],
+    ax: Axes,
+    cmap_params: CmapParams,
+    color_params: Colormap,
+    categorical: bool,
+    seg_erosionpx: int | None = None,
+    seg_boundaries: bool = False,
+    na_color: str | Tuple[float, ...] = (0, 0, 0, 0),
+    **kwargs: Any,
+) -> Tuple[Axes, Collection]:
+
+    img = _map_color_seg(
+        seg=seg,
+        cell_id=cell_id,
+        color_vector=color_vector,
+        color_source_vector=color_source_vector,
+        cmap_params=cmap_params,
+        seg_erosionpx=seg_erosionpx,
+        seg_boundaries=seg_boundaries,
+        na_color=na_color,
+    )
+
+    _cax = ax.imshow(
+        img,
+        rasterized=True,
+        cmap=cmap_params.cmap if not categorical else None,
+        norm=cmap_params.norm if not categorical else None,
```

Comment:
Why do both `_map_color_seg` and `imshow` get `cmap` and `norm`? Shouldn't you not need these for `imshow` since you're directly passing it an RGBA array?

Reply:
it's needed for the colorbar and the legend.